### ISSUE TITLE
Design question

### ISSUE THREAD

**Issue by jvmlet** [ctyar/Swashbuckle.Yarp#3]
Why application needs to know it's running behind YARP   and call 
````c#
 app.UseSwagger(o =>
        {
            o.AddYarp("/api1"); // ????
        });
````
The idea of reverse  proxy  is transparency , **nor** client **neither** server knows they are running with proxy in the middle...

**Comment by ctyar** [comment]
I agree with you that it would be the ideal solution.
However, the problem is that YARP has to add the path prefix (or any transform) on the fly on the swagger.json. This means that it has to read and rewrite it, which I don’t have a good solution for at the moment.

**Comment by jvmlet** [comment]
Turn do this kind of transportation in the app hosting the yarp. The ocelot  swagger does exactly this.

**Comment by ctyar** [comment]
Yes, that's where the logic would be ideally. It's just it would be a lot of work and right now I'm focused on other stuff on my free time, but I have it in the back of my mind.
